Classes/KIFUITestActor.m

```diff
@@ -676,18 +682,167 @@ - (void)selectPickerViewRowWithTitle:(NSString *)title inComponent:(NSInteger)co
+        __kifDateFormatter = [[NSDateFormatter alloc] init];
+    });
+    
+    return __kifDateFormatter;
+}
+
+// uses the date pickers locale to find the ordering of day month and year.
++ (NSArray<NSString *>*)dateOrderForDatePicker:(UIDatePicker *)datePicker
+{
+    NSString *dateFormatString = [NSDateFormatter dateFormatFromTemplate:@"yMMMMd" options:0 locale:datePicker.locale];
+    __block BOOL hasAddedYear = NO;
+    __block BOOL hasAddedMonth = NO;
+    __block BOOL hasAddedDay = NO;
+    NSMutableArray *dateOrder = [NSMutableArray array];
+    [dateFormatString enumerateSubstringsInRange:NSMakeRange(0, dateFormatString.length)
+                                         options:NSStringEnumerationByComposedCharacterSequences
+                                      usingBlock:^(NSString * _Nullable substring, NSRange substringRange, NSRange enclosingRange, BOOL * _Nonnull stop) {
+        if([substring containsString:@"y"] && !hasAddedYear) {
+            hasAddedYear = YES;
+            [dateOrder addObject:@"y"];
+        }
+        
+        if([substring containsString:@"d"] && !hasAddedDay) {
+            hasAddedDay = YES;
+            [dateOrder addObject:@"d"];
+        }
+        
+        if([substring containsString:@"M"] && !hasAddedMonth) {
+            hasAddedMonth = YES;
+            [dateOrder addObject:@"M"];
+        }
+    }];
+    
+    return [dateOrder copy];
+}
+
 - (void)selectDatePickerValue:(NSArray *)datePickerColumnValues
 {
-    [self selectPickerValue:datePickerColumnValues fromPicker:nil pickerType:KIFUIDatePicker withSearchOrder:KIFPickerSearchForwardFromStart];
+    [self selectDatePickerValue:datePickerColumnValues withSearchOrder:KIFPickerSearchForwardFromStart];
 }
+
 - (void)selectDatePickerValue:(NSArray *)datePickerColumnValues withSearchOrder:(KIFPickerSearchOrder)searchOrder
 {
-    [self selectPickerValue:datePickerColumnValues fromPicker:nil pickerType:KIFUIDatePicker withSearchOrder:searchOrder];
+    UIDatePicker *datePicker = [[[[UIApplication sharedApplication] datePickerWindow] subviewsWithClassNamePrefix:@"UIDatePicker"] firstObject];
+    [self selectDatePickerValue:datePickerColumnValues fromPicker:datePicker withSearchOrder:KIFPickerSearchForwardFromStart];
 }
 
-- (void)selectDatePickerValue:(NSArray *)datePickerColumnValues fromPicker:(UIPickerView *)picker withSearchOrder:(KIFPickerSearchOrder)searchOrder
+- (void)selectDatePickerValue:(NSArray *)datePickerColumnValues fromPicker:(UIDatePicker *)datePicker withSearchOrder:(KIFPickerSearchOrder)searchOrder
 {
-    [self selectPickerValue:datePickerColumnValues fromPicker:picker pickerType:KIFUIDatePicker withSearchOrder:searchOrder];
+    NSDateComponents *dateComponents = [[NSDateComponents alloc] init];
+    NSArray *dateOrder = [self.class dateOrderForDatePicker:datePicker];
+    
+    // Backwards compatibility while migrating away from this deprecated API.
+    if(datePicker.datePickerMode == UIDatePickerModeDate) {
```

Comment:
Should this logic take `datePickerStyle` into consideration for how to behave? It looks like wheel mode isn't gone in iOS 14, it just isn't the default.

Reply:
It doesn't care with the new way we do date picking, it works with any picker mode

Reply:
So my thinking is just move all types over to this